src/main/java/hudson/util/VersionNumber.java

```diff
@@ -513,16 +513,26 @@ public boolean isNewerThan(VersionNumber rhs) {
         return compareTo(rhs) > 0;
     }
 
+    /**
+     * Returns a digit (numeric component) by its position. Once a non-numeric component is found all remaining components
+     * are also considered non-numeric by this method.
+     *
+     * @param idx Digit position we want to retrieve starting by 1 (not zero).
```

Comment:
It was starting from zero before AIUI. I really think it would be safer to just deprecate this method and start with a new method where you can define the semantics you need.